js/src/sdk/utils/fileUtils.ts

```diff
@@ -10,11 +6,21 @@ import { COMPOSIO_DIR, TEMP_FILES_DIRECTORY_NAME } from "./constants";
  * @returns The path to the Composio directory.
  */
 export const getComposioDir = (createDirIfNotExists: boolean = false) => {
-  const composioDir = path.join(os.homedir(), COMPOSIO_DIR);
-  if (createDirIfNotExists && !fs.existsSync(composioDir)) {
-    fs.mkdirSync(composioDir, { recursive: true });
+  try {
```

Comment:
While the browser compatibility changes are good, consider using a more robust approach for handling Node.js specific modules:

```typescript
const isNode = typeof process !== 'undefined' && 
  process.versions != null && 
  process.versions.node != null;

const getFs = () => {
  if (!isNode) return null;
  try {
    return require('fs');
  } catch {
    return null;
  }
};
```

This would provide more reliable environment detection and cleaner error handling.